config.go

```diff
@@ -204,6 +204,7 @@ type Config struct {
 	ExternalIPs       []net.Addr
 	DisableListen     bool          `long:"nolisten" description:"Disable listening for incoming peer connections"`
 	DisableRest       bool          `long:"norest" description:"Disable REST API"`
+	DisableTLS        bool          `long:"notls" description:"Disable TLS for RPC connections"`
```

Comment:
We discussed this offline a bit. Just letting the user disable TLS is quite a big safety risk. We don't even let users disable macaroons if they're not at least listening on a private/loopback IP address. Maybe we need to add a condition for TLS in `lncfg.EnforceSafeAuthentication` as well?

Reply:
Also the description should say "for RPC and REST connections".

Reply:
Please do what @guggero suggest. As explained here https://github.com/lightningnetwork/lnd/pull/4648/files#r495456714 :

1. I have a very valid reason to bind on 0.0.0.0
2. Don't waste time of people silently changing the settings they meant.

Reply:
Added `EnforceSafeAuthentication` check for TLS also.

Note that this disallows disabling TLS and macaroons when listening on `0.0.0.0`, as you will have to specify a non-public address to listen on.

Reply:
Nice! Can confirm this works in a docker setup, see https://github.com/lightningnetwork/lnd/pull/4648#pullrequestreview-499285729.